src/main/java/org/jenkinsci/plugins/oic/OicAlgorithmValidator.java

```diff
@@ -0,0 +1,145 @@
+        jweSupportedAlgorithms.addAll(RSACryptoProvider.SUPPORTED_ALGORITHMS);
+        // RSA1_5 is deprecated and not a compliant algorithm.
+        jweSupportedAlgorithms.remove(JWEAlgorithm.RSA1_5);
+        jweSupportedAlgorithms.addAll(ECDHCryptoProvider.SUPPORTED_ALGORITHMS);
+        jweSupportedAlgorithms.addAll(PasswordBasedCryptoProvider.SUPPORTED_ALGORITHMS);
+
+        // Init complaint EncryptionMethods
+        supportedEncryptionMethod.addAll(ContentCryptoProvider.SUPPORTED_ENCRYPTION_METHODS);
+    }
+
+    /**
+     * Checks if the Jws signing algorithm used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isJwsAlgorithmFipsNonCompliant(Algorithm algorithm) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && algorithm != null) {
+            matchNotFound = jwsSupportedAlgorithms.stream().noneMatch(jwsAlgorithm -> jwsAlgorithm.equals(algorithm));
+        }
+        return matchNotFound;
+    }
+
+    /**
+     *  Checks if the Jwe encryption algorithm used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isJweAlgorithmFipsNonCompliant(Algorithm algorithm) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && algorithm != null) {
+            matchNotFound = jweSupportedAlgorithms.stream().noneMatch(jweAlgorithm -> jweAlgorithm.equals(algorithm));
+        }
+        return matchNotFound;
+    }
+
+    /**
+     *  Filter FIPS non-compliant Jwe encryption algorithm used for OIC configuration.
+     */
+    public static void filterFipsNonCompliantJweAlgorithm(List<JWEAlgorithm> algorithms) {
+        filterFipsNonCompliantAlgorithms(algorithms, OicAlgorithmValidator::isJweAlgorithmFipsNonCompliant);
+    }
+
+    /**
+     *  validate FIPS non-compliant Jwe encryption algorithm used for OIC configuration.
+     */
+    public static boolean isJwsAlgoFipsNonCompliant(List<JWSAlgorithm> algorithm) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && algorithm != null) {
+            for (JWSAlgorithm jwsAlgorithm : algorithm) {
+                matchNotFound = isJwsAlgorithmFipsNonCompliant(jwsAlgorithm);
+                if (matchNotFound) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     *  Filter FIPS non-compliant Jws encryption algorithm used for OIC configuration.
+     */
+    public static void filterFipsNonCompliantJwsAlgorithm(List<JWSAlgorithm> algorithms) {
+        filterFipsNonCompliantAlgorithms(algorithms, OicAlgorithmValidator::isJwsAlgorithmFipsNonCompliant);
+    }
+
+    /**
+     *  Checks if the encryption method used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isEncryptionMethodFipsNonCompliant(Algorithm encryptionMethod) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && encryptionMethod != null) {
+            if (!supportedEncryptionMethod.isEmpty()) {
```

Comment:
Can it be empty? IINW it's final and filled in the static block

Reply:
Done